genienlp/data_utils/example.py

```diff
@@ -64,19 +69,18 @@ def flatten(self):
                 result += field_val
         return result
 
-
-VALID_FEATURE_FIELDS = tuple(Feature.__annotations__.keys())
+    @staticmethod
+    def get_pad_entity(max_features_size):
+        # return None if not using NED
+        pad_feature = Entity()
+        for i, field in enumerate(VALID_FEATURE_FIELDS):
+            setattr(pad_feature, field, [0] * max_features_size)
+        return pad_feature
 
 
-def get_pad_feature(feature_fields, ned_features_default_val, ned_features_size):
-    # return None if not using NED
-    pad_feature = None
-    if len(feature_fields):
-        pad_feature = Feature()
-        for i, field in enumerate(feature_fields):
-            assert field in VALID_FEATURE_FIELDS
-            setattr(pad_feature, field, [ned_features_default_val[i]] * ned_features_size[i])
-    return pad_feature
+# 1: to remove self
+signature = inspect.signature(Entity.__init__)
+VALID_FEATURE_FIELDS = tuple(signature.parameters.keys())[1:]
```

Comment:
Honestly, this would be more readable by writing explicitly `["type_id", "type_prob", "qid"]`. It's three fields, and defined right above it.